Src/Sarif.PatternMatcher/Fingerprint.cs

```diff
@@ -157,91 +160,91 @@ internal static string ToString(Fingerprint f, ISet<string> denyList)
 
             if (!string.IsNullOrEmpty(f.PersonalAccessToken) && !denyList.Contains(PersonalAccessTokenKeyName))
             {
-                components.Add($"[{PersonalAccessTokenKeyName}={f.PersonalAccessToken.Trim()}]");
+                components.Add(PersonalAccessTokenKeyName, $"[{PersonalAccessTokenKeyName}={f.PersonalAccessToken.Trim()}]");
             }
 
             if (!string.IsNullOrEmpty(f.Platform) && !denyList.Contains(PlatformKeyName))
             {
-                components.Add($"[{PlatformKeyName}={f.Platform.Trim()}]");
+                components.Add(PlatformKeyName, $"[{PlatformKeyName}={f.Platform.Trim()}]");
             }
 
             if (!string.IsNullOrEmpty(f.Port) && !denyList.Contains(PortKeyName))
             {
-                components.Add($"[{PortKeyName}={f.Port.Trim()}]");
+                components.Add(PortKeyName, $"[{PortKeyName}={f.Port.Trim()}]");
             }
 
             if (!string.IsNullOrEmpty(f.Resource) && !denyList.Contains(ResourceKeyName))
             {
-                components.Add($"[{ResourceKeyName}={f.Resource.Trim()}]");
+                components.Add(ResourceKeyName, $"[{ResourceKeyName}={f.Resource.Trim()}]");
             }
 
             if (!string.IsNullOrEmpty(f.SasToken) && !denyList.Contains(SasTokenKeyName))
             {
-                components.Add($"[{SasTokenKeyName}={f.SasToken.Trim()}]");
+                components.Add(SasTokenKeyName, $"[{SasTokenKeyName}={f.SasToken.Trim()}]");
             }
 
             if (!string.IsNullOrEmpty(f.SymmetricKey128Bit) && !denyList.Contains(SymmetricKey128BitKeyName))
             {
-                components.Add($"[{SymmetricKey128BitKeyName}={f.SymmetricKey128Bit.Trim()}]");
+                components.Add(SymmetricKey128BitKeyName, $"[{SymmetricKey128BitKeyName}={f.SymmetricKey128Bit.Trim()}]");
             }
 
             if (!string.IsNullOrEmpty(f.SymmetricKey256Bit) && !denyList.Contains(SymmetricKey256BitKeyName))
             {
-                components.Add($"[{SymmetricKey256BitKeyName}={f.SymmetricKey256Bit.Trim()}]");
+                components.Add(SymmetricKey256BitKeyName, $"[{SymmetricKey256BitKeyName}={f.SymmetricKey256Bit.Trim()}]");
             }
 
             if (!string.IsNullOrEmpty(f.Thumbprint) && !denyList.Contains(ThumbprintKeyName))
             {
-                components.Add($"[{ThumbprintKeyName}={f.Thumbprint.Trim()}]");
+                components.Add(ThumbprintKeyName, $"[{ThumbprintKeyName}={f.Thumbprint.Trim()}]");
             }
 
             if (!string.IsNullOrEmpty(f.Uri) && !denyList.Contains(UriKeyName))
             {
-                components.Add($"[{UriKeyName}={f.Uri.Trim()}]");
+                components.Add(UriKeyName, $"[{UriKeyName}={f.Uri.Trim()}]");
             }
 
             return components.Count > 0 ?
-                string.Concat(components) :
+                string.Concat(components.OrderBy(c => c.Key).Select(v => v.Value)) :
```

Comment:
>components [](start = 30, length = 10)

Good change. We could have incorporated the IsNullOrEmptyCheck here as well?

Reply:
just added. what about my other question? what do u think?

---
In reply to: [593336660](https://github.com/microsoft/sarif-pattern-matcher/pull/289#discussion_r593336660) [](ancestors = 593336660)